src/vea/README.md

```diff
@@ -0,0 +1,28 @@
+# Versioned Block Allocator
+
+The Versioned Block Allocator is used by VOS for managing blocks on NVMe SSD, it's basically an extent based block allocator specially designed for DAOS.
+
+## Allocation metadata
+
+The blocks allocated by VEA are used to store single value or array value in VOS. Since the address and size from each allocation is tracked in VOS index trees, the VEA allocation metadata tracks only free extents in a btree, more importantly, this allocation metadata is stored on SCM along with the VOS index trees, so that block allocation and index updating could be easily made into single PMDK transaction, at the same time, the performance would be much better than storing the metadata on NVMe SSD.
```

Comment:
Minor optional cleanup:
1st sentence: "used to store **either** single **values** or array **values** in VOS.
2nd sentence: "address and size from each allocation **are** tracked", "made into **a** single PMDK transaction"